CredentialProvider/CredentialProviderPresenter.swift

```diff
@@ -0,0 +1,118 @@
+            cancelWith(.failed)
+        } else if let id = credentialIdentity.recordIdentifier {
+            
+            profile.logins.get(id: id).upon { [weak self] result in
+                switch result {
+                case .failure:
+                    self?.cancelWith(.failed)
+                case .success(let record):
+                    if let passwordCredential = record?.passwordCredential {
+                        self?.view?.extensionContext.completeRequest(withSelectedCredential: passwordCredential, completionHandler: nil)
+                    } else {
+                        self?.cancelWith(.userInteractionRequired)
+                    }
+                }
+            }
+        }
+    }
+    
+
+    func showCredentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
+    if self.profile.logins.reopenIfClosed() != nil {
+            cancelWith(.failed)
+        } else {
+            profile.logins.list().upon {[weak self] result in
+                switch result {
+                case .failure:
+                    self?.cancelWith(.failed)
+                case .success(let loginRecords):
+                    
+                    var sortedLogins = loginRecords.sorted(by: <)
+                    for (index, element) in sortedLogins.enumerated() {
+                        if let identifier = serviceIdentifiers.first?.identifier.asURL?.domainURL.absoluteString.titleFromHostname, element.passwordCredentialIdentity.serviceIdentifier.identifier.contains(identifier) {
+                            sortedLogins.remove(at: index)
+                            sortedLogins.insert(element, at: 0)
+                        }
+                    }
+                    
+                    let dataSource = sortedLogins.map { ($0.passwordCredentialIdentity, $0.passwordCredential) }
+                    DispatchQueue.main.async {
+                        self?.view?.show(itemList: dataSource)
+                    }
+                }
+            }
+        }
+    }
+    
+
+    func credentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
+        view?.showWelcome()
+        
+        guard let authInfo = KeychainWrapper.sharedAppContainerKeychain.authenticationInfo(), authInfo.requiresValidation() else {
+            showCredentialList(for: serviceIdentifiers)
+            return
+        }
+        
+        AppAuthenticator.presentAuthenticationUsingInfo(
+            authInfo,
+            touchIDReason: .AuthenticationLoginsTouchReason,
+            success: { self.showCredentialList(for: serviceIdentifiers)},
+            cancel: { self.cancelWith(.userCanceled) },
+            fallback: { [weak self] in
+                self?.view?.showPassword { isOk in
+                    if isOk { self?.showCredentialList(for: serviceIdentifiers) }
+                }
+            })
+    }
+    
+
+    func prepareAuthentication(for credentialIdentity: ASPasswordCredentialIdentity) { }
```

Comment:
Is this needed for a protocol conformance? Or can we remove it?

Reply:
This was not used so I removed it both here and the call to it in `CredentialProviderViewController.prepareInterfaceToProvideCredential`

Cc @razvanlitianu to make sure this wasn't here for a reason. I am pretty sure it was simply dead code.